astroscrappy/astroscrappy.pyx

```diff
@@ -246,22 +289,26 @@ def detect_cosmics(indat, inmask=None, float sigclip=4.5, float sigfrac=0.3,
         s = rebin(conved)
         del conved
 
-        # Build a the noise map, to compare the laplacian to
+        # Build a noise map, to compare the laplacian to, based on the input
+        # data or (if supplied) input variance. If variance is used, it still
+        # gets filtered, since we can best identify cosmic rays using what the
+        # noise in a pixel *should* be, rather than what it actually is after
+        # a cosmic ray hit.
         if sepmed:
-            m5 = sepmedfilt7(cleanarr)
+            m5 = sepmedfilt7(clean_var)
         else:
-            m5 = medfilt5(cleanarr)
+            m5 = medfilt5(clean_var)
 
         # Clip noise so that we can take a square root
         m5[m5 < 0.00001] = 0.00001
-        noise = np.sqrt(m5 + readnoise * readnoise)
-
-        if cleantype != 'median':
-            del m5
+        if bkg is not None:
+            m5 += bkg
```

Comment:
`bkg * gain` ?
Might be easier if `bkg` is multiplied by the gain at the beginning.